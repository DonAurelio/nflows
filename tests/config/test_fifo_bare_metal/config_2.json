{
    "dag_file": "./tests/workflows/test_fifo_bare_metal/config_2.dot",

    "scheduler_type": "fifo",
    "scheduler_params": [
        "fifo_prioritize_by_core_id=yes",
        "fifo_prioritize_by_exec_order=yes"
    ],

    "mapper_type": "bare-metal",
    "mapper_mem_policy_type": "default",
    "mapper_mem_bind_numa_node_ids": [],

    "core_avail_mask": "0x0F",
    "flops_per_cycle": 32,
    "clock_frequency_type": "static",
    "clock_frequency_hz": 1000000000,

    "distance_matrices": {
        "latency_ns": "./tests/system/test_fifo_bare_metal/2_lat.txt",
        "bandwidth_gbps": "./tests/system/test_fifo_bare_metal/2_bw.txt"
    },

    "out_file_name": "./tests/output/test_fifo_bare_metal/config_2.yaml"
}