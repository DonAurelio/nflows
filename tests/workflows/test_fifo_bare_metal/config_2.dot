digraph DataRedistribution {
    root    [size=2]; // Ignored in processing.
    end     [size=2]; // Ignored in processing.

    root -> Task_1  [size=2]; // Edge ignored.
    root -> Task_2  [size=2]; // Edge ignored.
    root -> Task_3  [size=2]; // Edge ignored.
    root -> Task_4  [size=2]; // Edge ignored.
    root -> Task_5  [size=2]; // Edge ignored.

    Task_1  [size=1000000000]; // 1Gflop
    Task_2  [size=1000000000];
    Task_3  [size=1000000000];
    Task_4  [size=1000000000];
    Task_5  [size=1000000000];

    Task_1 -> end   [size=2]; // Edge ignored.
    Task_2 -> end   [size=2]; // Edge ignored.
    Task_3 -> end   [size=2]; // Edge ignored.
    Task_4 -> end   [size=2]; // Edge ignored.
    Task_5 -> end   [size=2]; // Edge ignored.
}