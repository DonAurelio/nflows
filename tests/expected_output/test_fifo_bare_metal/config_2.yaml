name_to_thread_locality:
  Task_5: {numa_id: 0, core_id: 0}
  Task_4: {numa_id: 0, core_id: 3}
  Task_3: {numa_id: 0, core_id: 2}
  Task_1: {numa_id: 0, core_id: 0}
  Task_2: {numa_id: 0, core_id: 1}
